docs/updating-components.md

```diff
@@ -0,0 +1,74 @@
+# How to update components?
```

Comment:
This title is ambiguous. As a user, it sounds to me like this guide explains how I can deploy newer versions for the components on my running cluster. We should make it clear that this document is intended for maintainers/contributors and possibly separate it from the user-facing docs somehow.